app/components/primer/beta/flash_banner.ts

```diff
@@ -0,0 +1,62 @@
+import {controller, target} from '@github/catalyst'
+
+@controller
+class FlashBannerElement extends HTMLElement {
+  @target root: HTMLElement
+  @target message: HTMLElement
+
+  private lastFocusedElement: HTMLElement
+
+  connectedCallback() {
+    this.lastFocusedElement = (document.activeElement as HTMLElement) || document.body
+
+    if (document.readyState.toString() === 'ready' || document.readyState.toString() === 'complete') {
+      this.announceMessage()
+    } else {
+      document.addEventListener('readystatechange', () => {
+        if (document.readyState.toString() === 'complete') {
+          this.announceMessage()
+        }
```

Comment:
One thing that I noticed in dotcom was that with VoiceOver especially, was that the announcements would get interrupted. I believe that in dotcom, there are a lot of scripts that are still loading on the page and that somehow ends up interrupting the VoiceOver announcements. As a result, I ended up needing to wait for the `load` state that indicates when all the scripts have loaded, and delay on that. Wondering if you'd ran into that.

Reply:
Right yeah, that makes sense. I haven't tested yet in dotcom, which I'm sure will uncover a number of issues 😅  I tried adding a `load` listener but it never fired 🤷 so that's why I ended up with this `readystatechange` approach.

Reply:
Here is the load state in [document-ready.ts](https://github.com/github/github/blob/master/app/assets/modules/github/document-ready.ts#L24) in dotcom. Is this what you tried? 

I wonder if this might take the 75% success rate to 100%. If we cannot, we could try increasing the delay to see if it improves the announcements being made.

Reply:
In a Slack convo with folks from web-systems, we determined the `readystatechange` code is equivalent to the logic in document-ready.ts. Increasing the delay to 500ms seemed to bring the success rate up to 100% 😄